nf_core/modules/list.py

```diff
@@ -77,31 +77,33 @@ def pattern_msg(keywords):
             except UserWarning as e:
                 log.error(e)
                 return ""
+
             # Get installed modules
             self.get_pipeline_modules()
-            modules = self.module_names
 
-            # Filter the modules by keywords
-            modules = [mod for mod in modules if all(k in mod for k in keywords)]
+            # Filter by keywords
+            repos_with_mods = {
+                repo_name: [mod for mod in self.module_names[repo_name] if all(k in mod for k in keywords)]
+                for repo_name in self.module_names
+            }
 
             # Nothing found
-            if len(modules) == 0:
+            if sum(map(len, repos_with_mods)) == 0:
                 log.info(f"No nf-core modules found in '{self.dir}'{pattern_msg(keywords)}")
                 return ""
 
+            table.add_column("Repository")
+            table.add_column("Version SHA")
+            table.add_column("Message")
+            table.add_column("Date")
+
+            # Load 'modules.json'
             modules_json = self.load_modules_json()
-            if not modules_json:
-                # If the modules.json file is missing we show the old version
-                # of the list command, i.e only names
-                for mod in sorted(modules):
-                    table.add_row(mod)
-            else:
-                table.add_column("Version SHA")
-                table.add_column("Message")
-                table.add_column("Date")
 
+            for repo_name, modules in sorted(self.module_names.items(), key=lambda x: x[0]):
+                repo_entry = modules_json["repos"].get(repo_name, {})
                 for module in sorted(modules):
-                    module_entry = modules_json["repos"][self.modules_repo.name].get(module)
+                    module_entry = repo_entry.get(module)
```

Comment:
Does `.get()` return `False`/`None` when the key is not found?
I would have assumed this would give a key error. You could do `.get(module, None)` instead I think but you probably tested it already